README.md

```diff
@@ -15,13 +15,94 @@ refer to the documentation below for how to do that.
 ### Build requirements
  - a C++11 compiler (clang/gcc 4.7)
  - gnumake
- - cmake
- - boost
+ - CMake
+ - Boost
  - ruby (2.0 and newer)
- - [leatherman][leatherman](0.3.5 or newer)
+ - [leatherman][leatherman] (0.3.5 or newer)
  - [cpp-pcp-client][cpp-pcp-client] (master)
 
-Build with make and make install
+Initial Setup
+-------------
+
+Note: Testing custom facts requires Ruby 1.9+ with libruby built as a dynamic library; that often implies development builds of Ruby.
+
+### Setup on Fedora 23
+
+The following will install all required tools and libraries:
+
+    yum install boost-devel openssl-devel gcc-c++ make wget tar cmake
+
+### Setup on Mac OSX El Capitan (homebrew)
+
+This assumes Clang is installed and the system OpenSSL libraries will be used.
+
+The following will install all required libraries:
+
+    brew install cmake boost
+
+### Setup on Ubuntu 15.10 (Trusty)
+
+The following will install most required tools and libraries:
+
+    apt-get install build-essential libboost-all-dev libssl-dev wget tar cmake
+
+### Setup on Windows
```

Comment:
I'd be tempted to base it off something like https://github.com/puppetlabs/leatherman/pull/116/files instead, it provides much easier bootstrapping.